lib/pliny/middleware/canonical_log_line.rb

```diff
@@ -0,0 +1,121 @@
+module Pliny::Middleware
+  # Emits a "canonical log line", i.e. a single log line that contains as much
+  # relevant information about a request as possible and which makes for a
+  # single convenient reference point to understand all the vitals of any
+  # single request.
+  #
+  # This default implementation contains some useful data to get a project
+  # started, but it's usually recommended to vendor this middleware into your
+  # project and start adding some custom fields. Some examples of those might
+  # be:
+  #
+  #     * ID and email of an authenticated user.
+  #     * ID of API key used, OAuth application and scope.
+  #     * Remaining and total rate limits.
+  #     * Name of the service, HEAD revision, release number.
+  #     * Name of the internal system that initiated the request.
+  #
+  class CanonicalLogLine
+    class CanonicalLogLine
```

Comment:
Hah yeah, understandably.

I had the middleware called `CanonicalLogLineEmitter` until I renamed it at the last second to just `CanonicalLogLine` because it seemed to fit better into our existing middleware naming scheme.

I renamed the inner model to `LogLine` and annotated it with a comment in 49a084c. Hopefully this will help a little bit.